File: app/src/main/java/com/pietrantuono/podcasts/addpodcast/model/pojos/SinglePodcast.java
package com.pietrantuono.podcasts.addpodcast.model.pojos;

import android.os.Parcelable;

public interface SinglePodcast extends Parcelable {
    String getWrapperType();

    void setWrapperType(String wrapperType);

    String getKind();

    void setKind(String kind);

    Integer getCollectionId();

    void setCollectionId(Integer collectionId);

    Integer getTrackId();

    void setTrackId(Integer trackId);

    String getArtistName();

    void setArtistName(String artistName);

    String getCollectionName();

    void setCollectionName(String collectionName);

    String getTrackName();

    void setTrackName(String trackName);

    String getCollectionCensoredName();

    void setCollectionCensoredName(String collectionCensoredName);

    String getTrackCensoredName();

    void setTrackCensoredName(String trackCensoredName);

    String getCollectionViewUrl();

    void setCollectionViewUrl(String collectionViewUrl);

    String getFeedUrl();

    void setFeedUrl(String feedUrl);

    String getTrackViewUrl();

    void setTrackViewUrl(String trackViewUrl);

    String getArtworkUrl30();

    void setArtworkUrl30(String artworkUrl30);

    String getArtworkUrl60();

    void setArtworkUrl60(String artworkUrl60);

    String getArtworkUrl100();

    void setArtworkUrl100(String artworkUrl100);

    Double getCollectionPrice();

    void setCollectionPrice(Double collectionPrice);

    Double getTrackPrice();

    void setTrackPrice(Double trackPrice);

    Integer getTrackRentalPrice();

    void setTrackRentalPrice(Integer trackRentalPrice);

    Integer getCollectionHdPrice();

    void setCollectionHdPrice(Integer collectionHdPrice);

    Integer getTrackHdPrice();

    void setTrackHdPrice(Integer trackHdPrice);

    Integer getTrackHdRentalPrice();

    void setTrackHdRentalPrice(Integer trackHdRentalPrice);

    String getReleaseDate();

    void setReleaseDate(String releaseDate);

    String getCollectionExplicitness();

    void setCollectionExplicitness(String collectionExplicitness);

    String getTrackExplicitness();

    void setTrackExplicitness(String trackExplicitness);

    Integer getTrackCount();

    void setTrackCount(Integer trackCount);

    String getCountry();

    String getPrimaryGenreName();

    String getArtworkUrl600();
}

File: app/src/main/java/com/pietrantuono/podcasts/addpodcast/model/pojos/SinglePodcast.kt
package com.pietrantuono.podcasts.addpodcast.model.pojos

import android.os.Parcelable

interface SinglePodcast : Parcelable {
    var wrapperType: String

    var kind: String

    var collectionId: Int?

    var trackId: Int?

    var artistName: String

    var collectionName: String

    var trackName: String

    var collectionCensoredName: String

    var trackCensoredName: String

    var collectionViewUrl: String

    var feedUrl: String

    var trackViewUrl: String

    var artworkUrl30: String

    var artworkUrl60: String

    var artworkUrl100: String

    var collectionPrice: Double?

    var trackPrice: Double?

    var trackRentalPrice: Int?

    var collectionHdPrice: Int?

    var trackHdPrice: Int?

    var trackHdRentalPrice: Int?

    var releaseDate: String

    var collectionExplicitness: String

    var trackExplicitness: String

    var trackCount: Int?

    val country: String

    val currency: String

    val primaryGenreName: String

    val contentAdvisoryRating: String

    val artworkUrl600: String

    val genreIds: List<String>

    val genres: List<String>
}
